modules/swagger-codegen/src/main/java/io/swagger/codegen/CodegenOperation.java

```diff
@@ -30,6 +30,23 @@
     public List<Map<String, String>> examples;
     public ExternalDocs externalDocs;
 
+    private boolean nonempty(List<CodegenParameter> params)
+    {
+        return params != null && params.size() > 0;
+    }
+    public boolean getHasBodyParam() {
+        return nonempty(bodyParams);
+    }
+    public boolean getHasQueryParams() {
+        return nonempty(bodyParams);
+    }
+    public boolean getHasHeaderParams() {
+        return nonempty(bodyParams);
+    }
+    public boolean getHasPathParams() {
+        return nonempty(bodyParams);
+    }
+
```

Comment:
@DavidBiesack thanks for the PR to improve the HTML documentation. Instead of creating the function above to see if there's any body/query/header/path parameters, have you considered using `.length` provided by mustache template? 

Here is an example:
```
    {{#queryParams.length}}
    List<Pair> {{localVariablePrefix}}queryParams = new ArrayList<Pair>();
    {{/queryParams.length}}
```
The queryParameter variable would be declared once only if there's at least one query parameter.